### DISCUSSION TITLE
v12.24

### DISCUSSION THREAD

**Discussion by chrisws** [smallbasic/SmallBASIC#146] [General]
EMCC: Implemented emscripten web version
UI: add support for image.save("file.png") as per console
WEB: Added simple REST server support
COMMON: ARRAY now parses json with true/false correctly
COMMON: Fix #131 - Scalar * Vector doesnt work
COMMON: Fix array access with embedded strings #136
COMMON: Fixed hashmap access issue
COMMON: Fixed http_read to handle large HTTP headers
COMMON: Implemented window.setLocation(x,y) #102
COMMON: Removed 'Meaningless' CDBL, CINT, CREAL. These can be poly-filled with 'DEF' if required.
COMMON: Update plugin system to allow loading to be driven by the IMPORT statement
COMMON: m3Apply no longer crashes if the second argument is not an array
COMMON: parse JSON with SB ";" dimension syntax
CONSOLE: Added -i command switch for live mode
	This will run the BASIC program in a loop. It can be used with the debug module
	to rerun whenever the source code changes:

	import debug
	run("xdotool windowactivate `xdotool search --onlyvisible --name \"Emacs\"`")
	while (!debug.IsSourceModified())
	  rem main loop
	wend

<hr /><em>This discussion was created from the release <a href='https://github.com/smallbasic/SmallBASIC/releases/tag/'>v12.24</a>.</em>

**Comment by Joe7M** [comment]
Hi Chris,

I downloaded the AppImage for console and SDL. Both working fine in Manjaro. I like window.setLocation in combination with window.setSize(). Now only window.SetFullscreen() is missing for full control of the window. Maybe an idea for the next release. The import functionality to specify the path of the plugin is also great. Much easier to handle, than with the -m parameter.

When will you release the windows version?

Best regards, J7M

**Comment by Joe7M** [comment]
I just realized, that the source code file (tar.gz) for 12.24 in the release section can't be build because src/lib/jsm, lodepng, miniaudio and stb are empty.

**Reply by david-hot** [reply]
OH NO. That is the one I downloaded. Has it been fixed?
